static/app/components/inputGroup.tsx

```diff
@@ -0,0 +1,219 @@
+import {createContext, forwardRef, useContext, useEffect, useRef, useState} from 'react';
```

Comment:
I implemented this in its own file, so the import statement looks like:
```js
import {
  Input,
  InputGroup,
  InputLeadingItems,
  InputTrailingItems,
} from 'sentry/components/inputGroup';
```

Alternatively, I can just put everything inside `components/input.tsx`, so imports will look like:
```js
// current import statement:
// import Input from 'sentry/components/input';

// new version, if we just want the input:
import {Input} from 'sentry/components/input';

// if we also want the grouping:
import {
  Input,
  InputGroup,
  InputLeadingItems,
  InputTrailingItems,
} from 'sentry/components/input';
```

What do y'all think?